docs/en/install-upgrade/installing-stack-demo-secure.asciidoc

```diff
@@ -0,0 +1,894 @@
+----
++
+Respond to the command prompts as follows:
+
+* When asked if you want to generate a CSR, enter `n`.
+* When asked if you want to use an existing CA, enter `y`.
+* Provide the absolute path to your newly created CA certificate: `/etc/elasticsearch/certs/ca/ca.crt`.
+* Provide the absolute path to your newly created CA key: `/etc/elasticsearch/certs/ca/ca.key`.
+* Enter an expiration value for your certificate. You can enter the validity period in years, months, or days. For example, enter `1y` for one year.
+* When asked if you want to generate one certificate per node, enter `y`.
++
+You'll be guided through the creation of certificates for each node. Each certificate will have its own private key, and will be issued for a specific hostname or IP address.
+
+.. On separate lines, enter the hostname for your first {es} node, for example `mynode-es1`, and the IP address that clients can use to connect to your node.
++
+Note that this is the same value that's described in Step 2 of <<installing-stack-demo-self>>, for example `10.128.0.84`:
++
+["source","shell"]
+----
+mynode-es1
+10.128.0.84
+----
+.. When prompted, confirm that the settings are correct.
+.. Add the network IP address for the first {es} node (the same address you specified in the previous step):
++
+["source","shell"]
+----
+10.128.0.84
+----
+.. When prompted, confirm that the settings are correct.
+.. When prompted, choose to generate additional certificates, and then repeat the previous steps to add hostname and IP settings for each node in your {es} cluster.
+.. Provide a password for the generated `http.p12` keystore file.
+.. The generated files will be included in a zip archive. At the prompt, provide a path and filename for where the archive should be created.
++
+For this example we use: `/etc/elasticsearch/certs/elasticsearch-ssl-http.zip`:
++
+["source","shell"]
+----
+What filename should be used for the output zip file? [/usr/share/elasticsearch/elasticsearch-ssl-http.zip] /etc/elasticsearch/certs/elasticsearch-ssl-http.zip
+----
+
+. Earlier, when you generated the certificate for the transport layer, the default filename was `elastic-certificates.p12`. Now, when generating a certificate for the HTTP layer, the default filename is `http.p12`. This matches the name of the existing HTTP layer certificate file from when the initial {es} cluster was first installed.
++
+Just to avoid any possible name collisions, rename the existing `http.p12` file to distinguish it from the newly-created keystore:
++
+["source","shell"]
+----
+mv http.p12 http-old.p12
+----
+
+. Unzip the generated `elasticsearch-ssl-http.zip` archive:
++
+["source","shell"]
+----
+unzip elasticsearch-ssl-http.zip
+----
+
+. When the archive is unpacked, the certificate files are located in separate directories for each {es} node and for the {kib} node.
++
+You can run a recursive `ls` command to view the file structure:
++
+["source","shell"]
+----
+ls -lR elasticsearch kibana
```

Comment:
I think a more succint way to do it with full paths in one step would be: `ls -lR /usr/share/elasticsearch/{elasticsearch,kibana}`